graph/connected_components_with_dsu.cpp

```diff
@@ -0,0 +1,53 @@
+#include <iostream>
+#include <vector>
+#include <set>
+
+int N;  // denotes number of nodes;
+int parent[1000001];
```

Comment:
i think we can improve this too instead of creating array of this much large size you can dynamically allocate array of required size

Reply:
Thanks for pointing out. Fixed.
All arrays are now dynamically allocated.